### ISSUE TITLE
Issue with qi2labdatastore

### ISSUE THREAD

**Issue by mabbasi6** [QI2lab/merfish3d-analysis#32]
* wf-merfish version: f0444d0
* Python version: 3.10
* Operating System: Linux
* Branch: dataset-via-tensorstore

### Description

I am running the convert_to_qi2lab.py for iterative smFISH data and noticed it crashes. Here is the error I get:

`
Traceback (most recent call last):
  File "/home/momo/Repos/wf-merfish/pp-jobs/convert_to_qi2lab.py", line 216, in <module>
    convert_data()
  File "/home/momo/Repos/wf-merfish/pp-jobs/convert_to_qi2lab.py", line 99, in convert_data
    datastore.experiment_order = experiment_order
  File "/home/momo/Repos/wf-merfish/merfish3d-analysis/src/merfish3danalysis/qi2labDataStore.py", line 285, in experiment_order
    for idx in len(self._channels_in_data):
TypeError: 'int' object is not iterable
`

It might be that it needs to be `range(len(...))` in line 285 in order to loop.
@dpshepherd

**Comment by dpshepherd** [comment]
I thought this code already ran for you? 

Did you pull and try again?

**Comment by mabbasi6** [comment]
It did before. I pulled the latest update and got this error now.

**Comment by mabbasi6** [comment]
But previously, I had used the "convert_to_qi2lab.py" that was in the "human_olfactorybulb" folder of the commit version 324456a. I had made a copy of it and changed the dataset directory path.

**Comment by dpshepherd** [comment]
Ok, try now - i pushed a change. I haven't finish all of the tests for the new datastore yet, so it didn't catch this issue.

**Comment by mabbasi6** [comment]
It resolved the previous error but I get a new error:
`
'NoneType' object has no attribute 'dtype'
Could not access calibrations.zarr/noise_map
rounds:   0%|                                                                                             | 0/8 [00:00<?, ?it/scannot convert float NaN to integer                                                                       | 0/48 [00:00<?, ?it/s]
Error creating readout tile. Does it exist already?
rounds:   0%|                                                                                             | 0/8 [00:00<?, ?it/s]
Traceback (most recent call last):                                                                                              
  File "/home/momo/Repos/wf-merfish/pp-jobs/convert_to_qi2lab.py", line 216, in <module>
    convert_data()
  File "/home/momo/Repos/wf-merfish/pp-jobs/convert_to_qi2lab.py", line 137, in convert_data
    raw_image = imread(image_path)
  File "/home/momo/miniforge3/envs/wf-merfish/lib/python3.10/site-packages/tifffile/tifffile.py", line 1245, in imread
    with TiffFile(
  File "/home/momo/miniforge3/envs/wf-merfish/lib/python3.10/site-packages/tifffile/tifffile.py", line 4270, in __init__
    fh = FileHandle(file, mode=mode, name=name, offset=offset, size=size)
  File "/home/momo/miniforge3/envs/wf-merfish/lib/python3.10/site-packages/tifffile/tifffile.py", line 14659, in __init__
    self.open()
  File "/home/momo/miniforge3/envs/wf-merfish/lib/python3.10/site-packages/tifffile/tifffile.py", line 14678, in open
    self._fh = open(
FileNotFoundError: [Errno 2] No such file or directory: '/data/smFISH/20240827_Bartelle_MM_injurymodel/data_r0001_tile0000_2/data_r0001_tile0000_NDTiffStack.tif'
`
I notice that it looks for a directory with `data_r0001_tile0000_2` folder in it while the actual data folder is `data_r0001_tile0000_1`.

**Comment by dpshepherd** [comment]
Ok, that one you can fix locally - since it isn't in the datastore. I will fix it in the example (has to do with someone taking the first tile twice in one of the experiments).

**Comment by mabbasi6** [comment]
Thank you. It's running now, but also keeps printing this so I wanted to let you know:

`
annot convert float NaN to integer██████████▋                                                   | 19/48 [03:11<04:11,  8.68s/it]
Error creating readout tile. Does it exist already?
                                                                                                                               cannot convert float NaN to integer████████████▍                                                 | 20/48 [03:18<03:49,  8.21s/it]
Error creating readout tile. Does it exist already?
`

**Comment by dpshepherd** [comment]
Two things:
1. Since you already converted the data, you don't need to do it again
2. You can just run the updated register code with the previously converted data.

**Comment by dpshepherd** [comment]
Ok, I simplified some code and this shouldn't happen anymore with datastore creation. I tested with a few different datasets and it was fine. The error popped up due to a change in variable typing when parsing the experimental order.

I suggest that sometime soon you write your own `covert_to_qi2lab.py` just to make sure that you know what everything is doing. Happy to sit with you and go over each piece.
